Tchap/Modules/ForgotPassword/ForgotPasswordCoordinator.swift

```diff
@@ -141,25 +141,91 @@ final class ForgotPasswordCoordinator: ForgotPasswordCoordinatorType {
         let errorTitle: String = TchapL10n.errorTitleDefault
         let errorMessage: String
         
-        let nsError = error as NSError
-        
-        if let matrixErrorCode = nsError.userInfo[kMXErrorCodeKey] as? String, matrixErrorCode == kMXErrCodeStringThreePIDNotFound {
-            errorMessage = TchapL10n.forgotPasswordFormErrorEmailNotFound
+        if let forgotPwdError = error as? ForgotPasswordServiceError {
+            switch forgotPwdError {
+            case .invalidPassword(let reason):
+                switch reason {
+                case .tooShort(let minLength):
+                    errorMessage = TchapL10n.passwordPolicyTooShortPwdDetailedError(minLength)
+                case .no_digit:
+                    errorMessage = TchapL10n.passwordPolicyWeakPwdError
+                case .no_symbol:
+                    errorMessage = TchapL10n.passwordPolicyWeakPwdError
+                case .no_uppercase:
+                    errorMessage = TchapL10n.passwordPolicyWeakPwdError
+                case .no_lowercase:
+                    errorMessage = TchapL10n.passwordPolicyWeakPwdError
+                }
+            }
         } else {
-            errorMessage = TchapL10n.errorMessageDefault
+            let nsError = error as NSError
+            
+            if let matrixErrorCode = nsError.userInfo[kMXErrorCodeKey] as? String, matrixErrorCode == kMXErrCodeStringThreePIDNotFound {
+                errorMessage = TchapL10n.forgotPasswordFormErrorEmailNotFound
+            } else {
+                errorMessage = TchapL10n.errorMessageDefault
+            }
         }
         
         return ErrorPresentableImpl(title: errorTitle, message: errorMessage)
     }
     
+    private func shouldGoBackOnVerifyEmailError(_ error: Error) -> Bool {
+        let shouldGoBack: Bool
+        let nsError = error as NSError
+        
+        if let errCode = nsError.userInfo[kMXErrorCodeKey] as? String {
+            switch errCode {
+            case kMXErrCodeStringPasswordTooShort:
```

Comment:
You can group multiple cases on one line if you want: `case kMXErrCodeStringPasswordTooShort, kMXErrCodeStringPasswordNoDigit:`